sdk/cosmos/azure_data_cosmos/tests/framework/test_account.rs

```diff
@@ -0,0 +1,181 @@
+                _ => {}
+            }
+        }
+
+        let Some(endpoint) = account_endpoint else {
+            return Err("invalid connection string, missing 'AccountEndpoint'".into());
+        };
+
+        let Some(key) = account_key else {
+            return Err("invalid connection string, missing 'AccountKey'".into());
+        };
+
+        let context_id = format!(
+            "{}_{}_{}",
+            context.test_name(),
+            OffsetDateTime::now_utc().format(format_description!(
+                "[year]_[month]_[day]T[hour]_[minute]_[second]"
+            ))?,
+            Uuid::new_v4().as_simple()
+        );
+
+        TRACING.call_once(|| {
+            // Enable tracing for tests, if it's not already enabled
+            _ = tracing_subscriber::fmt()
+                .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+                .try_init();
+
+            // Ignore the failure. The most likely failure is that a global default trace dispatcher has already been set.
+            // And we don't want the tracing support to bring down the tests.
+        });
+
+        Ok(TestAccount {
+            context_id,
+            endpoint,
+            key,
+            options,
+        })
+    }
+
+    /// Create a [`CosmosClient`](azure_data_cosmos::CosmosClient) that connects to this account using a connection string.
+    #[cfg(feature = "key_auth")]
+    pub fn connect_with_key(
+        &self,
+        mut options: Option<CosmosClientOptions>,
+    ) -> Result<azure_data_cosmos::CosmosClient, Box<dyn std::error::Error>> {
+        let allow_invalid_certificates = match self.options.allow_invalid_certificates {
+            Some(b) => b,
+            None => std::env::var(ALLOW_INVALID_CERTS_ENV_VAR).map(|s| s.parse())??,
+        };
+
+        if allow_invalid_certificates {
+            let client = ClientBuilder::new()
+                .danger_accept_invalid_certs(true)
+                .pool_max_idle_per_host(0)
+                .build()?;
+            options = {
+                let mut o = options.unwrap_or_default();
+                o.client_options
+                    .set_transport(TransportOptions::new(Arc::new(client)));
+                Some(o)
+            };
+        }
+
+        Ok(azure_data_cosmos::CosmosClient::with_key(
+            &self.endpoint,
+            self.key.clone(),
+            options,
+        )?)
+    }
+
+    /// Generates a unique database ID including the [`TestAccount::context_id`].
+    ///
+    /// This database will be automatically deleted when [`TestAccount::cleanup`] is called.
+    pub fn unique_db(&self, base_id: &str) -> String {
+        format!("{}_{}", base_id, self.context_id)
+    }
+
+    /// Cleans up test resources, then drops the [`TestAccount`].
+    ///
+    /// Call this at the end of every test using the [`TestAccount`].
+    #[cfg(feature = "key_auth")]
+    pub async fn cleanup(self) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Q: Should I provide a `defer`-like `Drop` implementation on `TestContext`? For recorded tests, a `TestContext` is initialized at the beginning of every test and will be required to instrument clients for recording/playback, as well as adding sanitizers. Assuming captures work across async boundaries (or maybe we need some work here), we could do something like this:

```rust
#[recorded:test]
async fn test_foo(ctx: TestContext) -> Result<(), Box<dyn std::error::Error>> {
    let account = TestAccount::from_env(&ctx, ...);
    ctx.cleanup(async move || account.cleanup().await);

    // ...
}
```

Reply:
I was thinking of suggesting that! I'd like to be able to "automate" some degree of cleanup and actually started with a `Drop` implementation, but realized it needs async and that's kinda messy.

Reply:
One day maybe, perhaps, it might happen possibly: https://rust-lang.github.io/async-fundamentals-initiative/roadmap/async_drop.html

Reply:
Shall I go ahead and add that in this PR? I'm happy to do so, or just merge it and update it when it's added in a separate PR

Reply:
Ah, I just thought of a possible issue that makes this complicated enough to avoid for now: How should the cleanup run if the test panics? Do we need to [catch_unwind](https://doc.rust-lang.org/stable/std/panic/fn.catch_unwind.html)? What if the cleanup code panics (currently, if we didn't catch a cleanup panic, it would mask the actual test failure).